articles/contributing/api-design-principles.md

```diff
@@ -0,0 +1,426 @@
+---
+title: Q# API Design Principles
+description: Q# API Design Principles
+author: cgranade
+ms.author: chgranad
+ms.date: 10/12/2018
+ms.topic: article
+uid: microsoft.quantum.contributing.api-design
+---
+
+# Q# API Design Principles
+
+## General Principles
+
+- **DO** expose APIs that places the focus on quantum applications.
+  - **DON'T** expose APIs that focus primarily on low-level
+        implementation details.
+
+- **DO** choose operation and function names that reflect the
```

Comment:
Examples on this one might be good too?

Reply:
Actually, I think it would be great for most of these if there were examples in the QDK illustrating the **DO** cases it would be great to highlight them. Also just for visual clarity doing QDK examples in 
> boxes like this 

might be helpful :)

Reply:
I've refactored in a slightly different way, but I think it should work. Thanks for the suggestion!